root/static/scripts/edit/URLCleanup.js

```diff
@@ -2966,7 +2966,10 @@ const CLEANUPS: CleanupEntries = {
     match: [new RegExp('^(?:https?://)?(?:www\\.)?junodownload\\.com', 'i')],
     restrict: [LINK_TYPES.downloadpurchase],
     clean: function (url) {
-      return url.replace(/^https?:\/\/(?:www\.)?junodownload\.com\/([^?#]+).*$/, 'https://www.junodownload.com/$1');
+      url = url.replace(/^https?:\/\/(?:www\.)?junodownload\.com\/([^?#]+).*$/, 'https://www.junodownload.com/$1');
+      url = url.replace(/^https:\/\/www\.junodownload\.com\/(artists|labels)\/([^\/]+).*$/, 'https://www.junodownload.com/$1/$2/');
+      url = url.replace(/^https:\/\/www\.junodownload\.com\/products\/([^\/]+)\/([^\/]+).*$/, 'https://www.junodownload.com/products/$1/$2/');
+      return url;
     },
     validate: function (url, id) {
       const m = /^https:\/\/www\.junodownload\.com\/([a-z]+)\/.*$/.exec(url);
```

Comment:
The function `validate` should probably be updated to check the URL has been normalized too.